api/src/main/java/javax/enterprise/event/Event.java

```diff
@@ -100,6 +106,36 @@
 
     /**
      * <p>
+     * Fires an event asynchronously with the specified qualifiers and notifies synchronous and asynchronous observers.
+     * </p>
+     *
+     * @param event the event object
+     * @throws IllegalArgumentException if the runtime type of the event object contains a type variable
+     * @throws FireAsyncException   if a notified observer throws an exception, it will be caught and added
```

Comment:
From what I understand, CompletionStage doesn't fail, it allows to handle exception with handle() or exceptionally(). As describe in spec I think the last step in the async event pipeline building would be to check if we filled suppressedException in a FireAsyncExcetion created for the execution and if it's the case throw the exception. User will be able to handle this exception and dig thru the suppressed exceptions.